pipeline/data_store.py

```diff
@@ -823,10 +871,22 @@ def get_reference(self, provenance=None, session=None):
             The reference image for this image, or None if no reference is found.
 
         """
-        if self.ref_image is None:
-            with SmartSession(session, self.session) as session:
-                image = self.get_image(session=session)
-
+        with SmartSession(session, self.session) as session:
+            image = self.get_image(session=session)
+
+            if self.reference is not None:
+                if not (
+                        (self.reference.validity_start is not None or
+                         self.reference.validity_start <= image.observation_time) and
+                        (self.reference.validity_end is not None or
+                         self.reference.validity_end >= image.observation_time) and
+                        self.reference.filter == image.filter and
+                        self.reference.target == image.target and
```

Comment:
Usual comment: we should remove this, as there may well be different "targets" defined for the same point on space, and that will use the same reference.

Reply:
yes. I will make sure to round these up in a separate PR.